src/shared/Microsoft.Git.CredentialManager.Tests/GitConfigurationTests.cs

```diff
@@ -154,11 +155,82 @@ public void GitConfiguration_TryGet_Name_DoesNotExists_ReturnsFalse()
             IGitConfiguration config = git.GetConfiguration();
 
             string randomName = $"{Guid.NewGuid():N}.{Guid.NewGuid():N}";
-            bool result = config.TryGet(randomName, out string value);
+            bool result = config.TryGet(randomName, false, out string value);
             Assert.False(result);
             Assert.Null(value);
         }
 
+        [Fact]
+        public void GitConfiguration_TryGet_IsPath_True_ReturnsCanonicalPath()
+        {
+            string repoPath = CreateRepository(out string workDirPath);
+            ExecGit(repoPath, workDirPath, "config --local example.path ~/test").AssertSuccess();
+
+            string homeDirectory = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+            string gitPath = GetGitPath();
+            var trace = new NullTrace();
+            var git = new GitProcess(trace, gitPath, repoPath);
+            IGitConfiguration config = git.GetConfiguration();
+
+            bool result = config.TryGet("example.path", true, out string value);
+            Assert.True(result);
+            Assert.NotNull(value);
+            Assert.Equal(Path.GetFullPath($"{homeDirectory}/test"), Path.GetFullPath(value));
```

Comment:
Do we want to expand `value` to the full path here, or do we actually expect Git to do that for us already (i.e. use `value` here instead of `Path.GetFullPath(value)`)?

Reply:
I added the `Path.GetFullPath(value)` because, on Windows, it will return a path formatted with forward slashes, whereas `Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)` is formatted with backslashes. To your point, though, we should be testing `value` directly - I updated the implementation to explicitly replace the backslashes in `homeDirectory`, so the values should match.